deploy/bootstrap/run.sh

```diff
@@ -154,60 +202,70 @@ cat > ${PRIVATE}/conf.ini <<EOF
 log = debug
 #log = silent
 
-EOF
-if [[ $KEYSERVER == 'ega' ]]; then
-cat >> ${PRIVATE}/conf.ini <<EOF
-
 [keyserver]
 port = 8080
 
 [quality_control]
-keyserver_endpoint = http://keys:8080/keys/retrieve/%s/private/bin?idFormat=hex
+keyserver_endpoint = http://keys${HOSTNAME_DOMAIN}:8080/keys/retrieve/%s/private/bin?idFormat=hex
 
 [outgestion]
 # Just for test
-keyserver_endpoint = http://keys:8080/keys/retrieve/%s/private/bin?idFormat=hex
+keyserver_endpoint = http://keys${HOSTNAME_DOMAIN}:8080/keys/retrieve/%s/private/bin?idFormat=hex
 EOF
-else
-cat >> ${PRIVATE}/conf.ini <<EOF
-[keyserver]
-port = 8443
 
-[quality_control]
-keyserver_endpoint = https://keys:8443/retrieve/%s/private
+# Local broker connection
+MQ_CONNECTION_PARAMS=$(python -c "from urllib.parse import urlencode;                   \
+			          print(urlencode({ 'heartbeat': 0,                     \
+				                    'connection_attempts': 30,          \
+				                    'retry_delay': 10,                  \
+				                  }))")
+
+# Pika is not parsing the URL the way RabbitMQ likes.
+# So we add the parameters on the configuration file and
+# create the SSL socket ourselves
+# Some parameters can be passed in the URL, though.
+MQ_CONNECTION="amqps://${MQ_USER}:${MQ_PASSWORD}@mq${HOSTNAME_DOMAIN}:5671/%2F"
+
+# Database connection
+DB_CONNECTION_PARAMS=$(python -c "from urllib.parse import urlencode;                   \
+			          print(urlencode({ 'application_name': 'LocalEGA',     \
+				                    'sslmode': 'verify-full',           \
+				                    'sslcert': '/etc/ega/ssl.cert',     \
+				                    'sslkey': '/etc/ega/ssl.key.lega',  \
+				                    'sslrootcert': '/etc/ega/CA.cert',  \
+				                  }, safe='/-_.'))")
+
+DB_CONNECTION="postgres://lega_in:${DB_LEGA_IN_PASSWORD}@db${HOSTNAME_DOMAIN}:5432/lega"
 
-[outgestion]
-# Just for test
-keyserver_endpoint = https://keys:8443/retrieve/%s/private
-EOF
-fi
+#
+# Configuration file
+#
 
 cat >> ${PRIVATE}/conf.ini <<EOF
 
 ## Connecting to Local EGA
 [broker]
-host = mq
-username = ${MQ_USER}
-password = ${MQ_PASSWORD}
-connection_attempts = 30
-# delay in seconds
-retry_delay = 10
-
-[postgres]
-host = db
-port = 5432
-user = lega_in
-password = ${DB_LEGA_IN_PASSWORD}
-database = lega
+connection = ${MQ_CONNECTION}?${MQ_CONNECTION_PARAMS}
+
+enable_ssl = yes
+verify_peer = yes
+verify_hostname = no
+
+cacertfile = /etc/ega/CA.cert
+certfile = /etc/ega/ssl.cert
+keyfile = /etc/ega/ssl.key
+
+[db]
```

Comment:
These changes are not transferred to the `defaults.ini` file.
https://github.com/EGA-archive/LocalEGA/blob/6c8500d736000b3dc263f811280ce4f44c8b07a4/lega/conf/defaults.ini#L47
With the result that the result that the container will not start unless the bootstrap script is used.

The defaults.ini should contain a complete configuration example.

Reply:
Good catch. I updated the defaults.ini.